indico/modules/categories/controllers/display.py

```diff
@@ -670,3 +671,27 @@ def _get_event_data(self, event_query):
     def _render_ongoing_events(self, ongoing_events):
         template = get_template_module('categories/display/_calendar_ongoing_events.html')
         return template.render_ongoing_events(ongoing_events, self.category.display_tzinfo)
+
+
+class RHCategoryUpcomingEvent(RHDisplayCategoryBase):
+    """Redirect to the upcoming event of a category."""
+
+    def _process(self):
+        event_id = self._get_upcoming_event()
+        if event_id:
+            resp = redirect(url_for('events.display', confId=event_id))
+        else:
+            flash(_('There are no upcoming events for this category'))
+            resp = redirect(url_for('categories.display', category_id=self.category.id))
+        return resp
+
+    def _get_upcoming_event(self):
+        query = (Event.query
+                 .filter(Event.is_visible_in(self.category),
+                         Event.start_dt > now_utc())
+                 .options(load_only('id', 'start_dt'))
```

Comment:
You don't need to load the `start_dt`, but the access check needs some other columns so currently your will spam queries for each access check. It's better to just not restrict the columns except for performance-critical places (which this isn't).

However, to avoid query spam in to the access checks you probably want `.options(subqueryload('acl_entries'))`.